libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -50,8 +48,20 @@ public Task ContinueConversation(string botAppId, ConversationReference referenc
             if (callback == null)
                 throw new ArgumentNullException(nameof(callback)); 
 
-            var context = new BotFrameworkTurnContext(botAppId, this, reference.GetPostToBotMessage());
-            return RunPipeline(context, callback);
+            var context = new TurnContext(this, reference.GetPostToBotMessage());
+
+            // Hand craft Claims Identity.
+            ClaimsIdentity claimsIdentity = new ClaimsIdentity(new List<Claim>
+            {
+                // Adding claims for both Emulator and Channel.
+                new Claim(AuthenticationConstants.AudienceClaim, botAppId),
+                new Claim(AuthenticationConstants.AppIdClaim, botAppId)
+            });
+
+            context.Services.Add<IIdentity>(claimsIdentity);
```

Comment:
Is there really any value adding this to the services? You directly pass it to `CreateConnectorClientAsync`, so it's not like there's any code that's looking for it.

If you still think it should be registered, might I suggest you register it under an explicit, well known key (maybe something like `"BotIdentity"`) rather than as "the" `IIdentity` since one could imagine there being some conflict around that interface at some point.

Reply:
This is for extension SDKs to use. If you want to build a middleware for creating custom connector client (Teams extension SDK) or for creating other clients (e.g. Graph client or something).